tests/rhbz_functions.at

```diff
@@ -0,0 +1,275 @@
+
+        xmlrpc_struct_find_value(env, xml, member, &xml_member);
+        if (env->fault_occurred)
+        {
+            printf("Error searching for member %s: %s\n", member, env->fault_string);
+            return ret;
+        }
+        xmlrpc_read_string(env, xml_member, (const char **)&ret);
+        if (env->fault_occurred)
+        {
+            printf("Error reading string from member %s: %s\n", member, env->fault_string);
+            return ret;
+        }
+        xmlrpc_DECREF(xml_member);
+
+        return ret;
+}
+
+TS_MAIN
+{
+    pid_t pid = fork();
+
+    if (pid == 0)
+    {
+        int ret = 0;
+        char *argv[2];
+        argv[0] = (char *)"../../mock_bugzilla_server/mock_bugzilla_server.py";
+        argv[1] = NULL;
+        ret = execv(argv[0], argv);
+        if (ret == -1)
+        {
+            printf("execv failed: %s\n", strerror(errno));
+        }
+    }
+    else if (pid > 0)
+    {
+        libreport_g_verbose = 3;
+        char cwd_buf[PATH_MAX + 1];
+
+        static const char *dirs[] = {
+            NULL,
+            NULL,
+        };
+        dirs[0] = getcwd(cwd_buf, sizeof(cwd_buf));
+
+        static int dir_flags[] = {
+            CONF_DIR_FLAG_NONE,
+            -1,
+        };
+
+        unlink("libreport.conf");
+        FILE *lrf = fopen("libreport.conf", "wx");
+        assert(lrf != NULL);
+        fprintf(lrf, "AlwaysExcludedElements = maps, var_log_messages, proc_pid_status");
+        fclose(lrf);
+
+        assert(libreport_load_global_configuration_from_dirs(dirs, dir_flags));
+
+        struct bugzilla_struct {
+            char *b_login;
+            char *b_password;
+            const char *b_bugzilla_url;
+            const char *b_bugzilla_xmlrpc;
+            char *b_product;
+            char *b_product_version;
+            const char *b_DontMatchComponents;
+            int         b_ssl_verify;
+            int         b_create_private;
+            GList       *b_private_groups;
+        };
+
+        int bug_id = 0;
+
+        sleep(3);
+
+        g_autofree const char *token = NULL;
+        g_autofree const char *server_ret = NULL;
+        g_autofree const char *product_description = NULL;
+        struct bugzilla_struct rhbz = { 0 };
+        rhbz.b_bugzilla_url = "http://localhost:8080/";
+        rhbz.b_bugzilla_xmlrpc = g_build_filename(rhbz.b_bugzilla_url, "xmlrpc.cgi", NULL);
+        rhbz.b_ssl_verify = FALSE;
+
+        xmlrpc_env env;
+        xmlrpc_env_init(&env);
+        if (env.fault_occurred)
+            printf("Error in env_init: %s\n", env.fault_string);
+        xmlrpc_client_setup_global_const(&env);
+        if (env.fault_occurred)
+            printf("Error in setup_global_const: %s\n", env.fault_string);
+
+        struct abrt_xmlrpc *client = abrt_xmlrpc_new_client(rhbz.b_bugzilla_xmlrpc, rhbz.b_ssl_verify);
+
+        {
+            xmlrpc_value *result = abrt_xmlrpc_call_full(&env, client, "User.login", "{s:s,s:s}",
```

Comment:
I’m not sure we want to test these functions directly, because they are very thing wrappers for XMLRPC-C. Why not `rhbz_login()`?